pkg/infra/kubernetes/helm_chart.go

```diff
@@ -45,40 +45,20 @@ func (chart *HelmChart) KlothoConstructRef() []core.AnnotationKey { return chart
 func (chart *HelmChart) Id() string {
 	return fmt.Sprintf("%s:%s:%s", chart.Provider(), HELM_CHART_TYPE, chart.Name)
 }
-
-func (t *HelmChart) OutputTo(dest string) error {
-	errs := make(chan error)
-	files := t.Files
-	for idx := range files {
-		go func(f core.File) {
-			path := filepath.Join(dest, "charts", f.Path())
-			dir := filepath.Dir(path)
-			err := os.MkdirAll(dir, 0777)
-			if err != nil {
-				errs <- err
-				return
-			}
-			file, err := os.OpenFile(path, os.O_RDWR, 0777)
-			if os.IsNotExist(err) {
-				file, err = os.OpenFile(path, os.O_CREATE|os.O_WRONLY, 0777)
-			}
-			if err != nil {
-				errs <- err
-				return
-			}
-			_, err = f.WriteTo(file)
-			defer file.Close()
-			errs <- err
-		}(files[idx])
-	}
-
-	for i := 0; i < len(files); i++ {
-		err := <-errs
+func (t *HelmChart) GetOutputFiles() []core.File {
+	var outputFiles []core.File
+	for _, file := range t.Files {
+		buf := &bytes.Buffer{}
+		_, err := file.WriteTo(buf)
 		if err != nil {
-			return err
+			panic(err)
 		}
+		outputFiles = append(outputFiles, &core.RawFile{
+			FPath:   path.Join("charts", file.Path()),
+			Content: buf.Bytes(),
+		})
 	}
-	return nil
+	return outputFiles
```

Comment:
the path is different since we append it to the charts path. In the future we can clean up writing the files the first time so their path contains that and then change the helm chart template to not have it hardcoded

Reply:
👍 
Can you add a ticket or at least a TODO for the future clean up?

Reply:
https://github.com/orgs/klothoplatform/projects/1/views/18?pane=issue&itemId=26084072